### PR TITLE
Fix : awx.awx.group preserve hosts fails when there are no hosts

### LINKED ISSUE #11954: awx.awx.group option preserve hosts fails when there are no hosts.
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.

### Summary

when using awx.awx.groups the preserve hosts/groups options fails if there are no groups/hosts.

### AWX version

19.4.0

### Select the relevant components

- [ ] UI
- [ ] API
- [ ] Docs

### Installation method

minikube

### Modifications

no

### Ansible version

_No response_

### Operating system

_No response_

### Web browser

_No response_

### Steps to reproduce

```
    - name: Add host to temporary group for new vms
      awx.awx.group:
        name: new_vms
        inventory: "Temporary Inventory"
        hosts:
          - "{{ name }}"
        validate_certs: false
        preserve_existing_hosts: True
      when:
        - use_aap_inventory
        - vm is succeeded
```

### Expected results

It runs and adds the host to the empty group

### Actual results

```
TypeError: 'NoneType' object is not subscriptabl
```

### Additional information

Because [This line](https://github.com/ansible/awx/blob/593eebf062cd1f73c117502e0491f3c8532695a3/awx_collection/plugins/modules/group.py#L174) returns NoneType when there are no hosts in the group.

module needs to be updated to fix this issue.

### PR DESCRIPTION
**Pull request by darshanip**
##### SUMMARY
Fixes #11954 

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Collection

### CODE REVIEW

**Comment by fosterseth** [comment]
@sean-m-sullivan do you mind taking a look at this PR?

@darshanip do you mind adding a test case around this?  A new test case in this file https://github.com/ansible/awx/blob/66a3cb6b09501c6cbf08b3153291530c6af9f0f9/awx_collection/tests/integration/targets/group/tasks/main.yml#L72

File: awx_collection/plugins/modules/group.py [hunk 1]
```diff
@@ -172,7 +172,8 @@ def main():
         if (preserve_existing_hosts and relationship == 'hosts') or (preserve_existing_children and relationship == 'children'):
             preserve_existing_check = module.get_all_endpoint(group['related'][relationship])
```
Comment (sean-m-sullivan) [inline awx_collection/plugins/modules/group.py]: On testing where the Inventory exists, and the host exists, but nothing else. This section failed, put in PR to fix this on your branch
https://github.com/darshanip/awx/pull/1j

With both changes, it looks good!

**Review by sean-m-sullivan — APPROVED** [review]
Tested, and LGTM

**Comment by darshanip** [comment]
@fosterseth can someone verify test cases.

**Comment by rebeccahhh** [comment]
@fosterseth is this good to merge?

**Review by fosterseth — APPROVED** [review]

**Review by fosterseth — APPROVED** [review]

**Pull request merged**